File: src/main/java/ir/reservs/reservs/ui/splash/SplashContract.java
package ir.reservs.reservs.ui.splash;

import ir.reservs.reservs.ui.base.BaseContract;

class SplashContract {
    interface View {
        void openMainActivity();

        void openLoginActivity();

        void onError(String text);
    }

    interface Presenter extends BaseContract.BasePresenter<View> {

    }
}

File: src/main/java/ir/reservs/reservs/ui/splash/SplashContract.kt
package ir.reservs.reservs.ui.splash

import ir.reservs.reservs.ui.base.BaseFragmentContract

class SplashContract {
    interface View : BaseFragmentContract.View {
        fun openMainActivity()

        fun openLoginActivity()

    }

    interface Presenter : BaseFragmentContract.Presenter<View> {

    }
}